File: samples/src/commonMain/kotlin/sample/mixing/MixingAnnotations.kt
package sample.mixing

import deezer.kustomexport.KustomExport

@JsExport
open class JsExported {
    val data = 12
}

@KustomExport
open class KustomExported {
    val data = 12
}

@KustomExport
data class MixingAnnotations(val jsExported: JsExported, val kustomExported: KustomExported) {
    fun makeJ(): JsExported = JsExported()
    fun makeK(): KustomExported = KustomExported()
    fun consumeJ(j: JsExported) = "Consumed ${j::class} - ${j.data}"
    fun consumeK(k: KustomExported) = "Consumed ${k::class} - ${k.data}"
}

File: samples/src/commonMain/kotlin/sample/mixing/MixingAnnotations.ts
import { runTest } from "../shared_ts/RunTest"
import { assertEquals, assertEqualsQuiet } from "../shared_ts/Assert"
import { sample } from '@kustom/Samples'

runTest("MixingAnnotations", () : void => {
    var j = new sample.mixing.JsExported()
    assertEquals(12, j.data, "Retrieve data from a @JsExport class")
    var k = new sample.mixing.js.KustomExported()
    assertEquals(12, k.data, "Retrieve data from a @KustomExport class")

    var mixer = new sample.mixing.js.MixingAnnotations(j, k)
    var jj = mixer.makeJ()
    var kk = mixer.makeK()

    assertEqualsQuiet("Consumed class JsExported - 12", mixer.consumeJ(j), "Consumed JsExported from Kotlin")
    assertEqualsQuiet("Consumed class KustomExported - 12", mixer.consumeK(k), "Consumed KustomExported from Kotlin")
})
